### PR TITLE
Install Linkage Checker enforcer rule for the spring-cloud-gcp-dependencies BOM

### PR DESCRIPTION
**Pull request by suztomo**
This is a followup of #1658 

- Linkage Checker Enforcer rule 1.5.0
- Increase of JVM to 2GB heap
- Setting `reportOnlyReachable` to only see reachable linkage errors. (Otherwise you'll see 22,000 lines of linkage errors: https://gist.github.com/suztomo/b73c69e5b769108a62657bb84fd0203a)

### CODE REVIEW

**Comment by suztomo** [comment]
Travis failed at install task.

```
[INFO] --- maven-install-plugin:2.5.2:install (default-install) @ spring-cloud-gcp-pubsub-stream-binder-functional-sample-test ---
[INFO] Installing /home/travis/build/spring-cloud/spring-cloud-gcp/spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-stream-binder-functional-sample/spring-cloud-gcp-pubsub-stream-binder-functional-sample-test/target/spring-cloud-gcp-pubsub-stream-binder-functional-sample-test-1.2.5.BUILD-SNAPSHOT.jar to /home/travis/.m2/repository/org/springframework/cloud/spring-cloud-gcp-pubsub-stream-binder-functional-sample-test/1.2.5.BUILD-SNAPSHOT/spring-cloud-gcp-pubsub-stream-binder-functional-sample-test-1.2.5.BUILD-SNAPSHOT.jar
[INFO] Installing /home/travis/build/spring-cloud/spring-cloud-gcp/spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-bus-config-sample/spring-cloud-gcp-pubsub-bus-config-sample-server-local/pom.xml to /home/travis/.m2/repository/org/springframework/cloud/spring-cloud-gcp-pubsub-bus-config-sample-server-local/1.2.5.BUILD-SNAPSHOT/spring-cloud-gcp-pubsub-bus-config-sample-server-local-1.2.5.BUILD-SNAPSHOT.pom
[INFO] Installing /home/travis/build/spring-cloud/spring-cloud-gcp/spring-cloud-gcp-samples/spring-cloud-gcp-pubsub-stream-binder-functional-sample/spring-cloud-gcp-pubsub-stream-binder-functional-sample-test/pom.xml to /home/travis/.m2/repository/org/springframework/cloud/spring-cloud-gcp-pubsub-stream-binder-functional-sample-test/1.2.5.BUILD-SNAPSHOT/spring-cloud-gcp-pubsub-stream-binder-functional-sample-test-1.2.5.BUILD-SNAPSHOT.pom
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received
```

**Comment by elefeint** [comment]
That looks like a Travis flake; we get them once every couple of weeks.

**Comment by suztomo** [comment]
Codecov bot shows lots of red. I didn't change any Java code though. Is Linkage Checker enforcer rule interfering the code coverage retrieval?

Screen Shot 2020-08-13 at 11 07 43


@elefeint Thanks.

**Comment by suztomo** [comment]
Travis behaves strangely. 

```
Execution enforce-linkage-checker of goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce failed: The source class in the reference is no longer available in the class path: Exception while looking for class org.springframework.cloud.gcp.autoconfigure.sql.GcpCloudSqlProperties: java.io.IOException: Couldn't find: org/springframework/cloud/gcp/autoconfigure/sql/GcpCloudSqlProperties.class
```

**Comment by suztomo** [comment]
4GB also shows the problem. Let me dig further.

**Comment by elefeint** [comment]
Ignore codecov; its base is unit + integration tests, so all pull requests look like they drop coverage. We really need to have a unit-only base to make it a meaningful tool.

**Comment by suztomo** [comment]
@elefeint I see. Thank you for info.

Linkage checker gets stuck in Linux (even gLinux). In Mac, it works fine. Closing this for now.  I'll investigate further.

File: .travis.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ script:
       ./mvnw test -B -P codecov ${INTEGRATION_TEST_FLAGS} && bash <(curl -s https://codecov.io/bash) -F integration;
     fi;
 install:
-  - ./mvnw -T 1.5C install -DskipTests=true -Dmaven.javadoc.skip=true -Dspring.profiles.active=spring -P full-checkstyle -B -V
+  - ./mvnw install -DskipTests=true -Dmaven.javadoc.skip=true -Dspring.profiles.active=spring -P full-checkstyle -B -V
```
Comment (suztomo) [inline .travis.yml:35]: In my test, this thread configuration is related to the bug. Does this work in Travis?

**Comment by suztomo** [comment]
Commit 1e6d143 passed Travis. So the threading parameter "-T 1.5C" is related to the error.

```
[INFO] --- maven-enforcer-plugin:3.0.0-M3:enforce (enforce-linkage-checker) @ spring-cloud-gcp-dependencies ---
[INFO] No reachable error found
```

File: .travis.yml (hunk 1 above)
Comment (suztomo) [reply .travis.yml:35]: It worked.

**Comment by suztomo** [comment]
It turned out that Linkage Checker cannot handle multi-thread Maven execution, because a JAR file is updated by another install task while it's running.

**Pull request closed without merging**